deepchecks/vision/checks/performance/robustness_report.py

```diff
@@ -0,0 +1,406 @@
+        # Iterate both datasets and compare results
+        baseline_sampler = iter(dataset.get_data_loader().dataset)
+        aug_sampler = iter(aug_dataset.get_data_loader().dataset)
+
+        # Validating on a single sample that the augmentation had affected
+        for (sample_base, sample_aug) in zip(baseline_sampler, aug_sampler):
+            # Skips any sample without label
+            label = sample_base[1]
+            if label is None or (isinstance(label, Sized) and len(label) == 0):
+                continue
+
+            if get_image_info(sample_base[0]).is_equals(sample_aug[0]):
+                raise DeepchecksValueError('Found that images have not been affected by adding augmentation. '
+                                           'This might be a problem with the implementation of Dataset.__getitem__')
+
+            # For classification does not check label for difference
+            if dataset.task_type != TaskType.CLASSIFICATION:
+                label_a = dataset.label_transformer([sample_base[1]])[0]
+                label_b = dataset.label_transformer([sample_aug[1]])[0]
+                if torch.equal(label_a, label_b):
+                    raise DeepchecksValueError('Found that labels have not been affected by adding augmentation. label '
+                                               f'before {label_a} and after {label_b}. This might be a problem with '
+                                               f'the implementation of Dataset.__getitem__')
+            # If all validations passed return
+            return
+
+    def _calc_top_affected_classes(self, base_results, augmented_results, dataset, n_classes_to_show):
+        def calc_percent(a, b):
+            return (a - b) / b if b != 0 else 0
+
+        aug_top_affected = defaultdict(list)
+        metrics = base_results['Metric'].unique().tolist()
+        for metric in metrics:
+            single_metric_scores = augmented_results[augmented_results['Metric'] == metric][['Class', 'Value']] \
+                .set_index('Class')
+            single_metric_scores['Base'] = base_results[base_results['Metric'] == metric][['Class', 'Value']] \
+                .set_index('Class')
+            diff = single_metric_scores.apply(lambda x: calc_percent(x.Value, x.Base), axis=1)
+
+            for index_class, diff_value in diff.sort_values().iloc[:n_classes_to_show].iteritems():
+                aug_top_affected[metric].append({'class': index_class,
+                                                 'value': single_metric_scores.at[index_class, 'Value'],
+                                                 'diff': diff_value,
+                                                 'samples': dataset.get_samples_per_class()[index_class]})
+        return aug_top_affected
+
+    def _calc_performance_diff(self, mean_base, augmented_metrics):
+        def difference(aug_score, base_score):
+            # TODO how to handle zero base score
```

Comment:
return max value? Not 0. Quick hack would be return 1. Better would be to look at how it's done in tabular.

Reply:
note it's not the metric value, it's the difference. so 0 ratio means no change from base to aug. the other option is base is 0 and aug is not, and then it means aug was better, which is less interesting probably just random luck